src/analysis/processing/qgsalgorithmcheckgeometrydegeneratepolygon.cpp

```diff
@@ -0,0 +1,194 @@
+/***************************************************************************
+                        qgsalgorithmcheckgeometrydegeneratepolygon.cpp
+                        ---------------------
+   begin                : January 2024
+   copyright            : (C) 2024 by Jacky Volpes
+   email                : jacky dot volpes at oslandia dot com
+***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmcheckgeometrydegeneratepolygon.h"
+#include "qgsgeometrycheckcontext.h"
+#include "qgsgeometrycheckerror.h"
+#include "qgsgeometrydegeneratepolygoncheck.h"
+#include "qgspoint.h"
+#include "qgsvectorlayer.h"
+#include "qgsvectordataproviderfeaturepool.h"
+
+///@cond PRIVATE
+
+QString QgsGeometryCheckDegeneratePolygonAlgorithm::name() const
+{
+  return QStringLiteral( "checkgeometrydegeneratepolygon" );
+}
+
+QString QgsGeometryCheckDegeneratePolygonAlgorithm::displayName() const
+{
+  return QObject::tr( "Check Geometry (Degenerate polygon)" );
+}
+
+QStringList QgsGeometryCheckDegeneratePolygonAlgorithm::tags() const
+{
+  return QObject::tr( "check,geometry,degeneratepolygon" ).split( ',' );
+}
+
+QString QgsGeometryCheckDegeneratePolygonAlgorithm::group() const
+{
+  return QObject::tr( "Check geometry" );
+}
+
+QString QgsGeometryCheckDegeneratePolygonAlgorithm::groupId() const
+{
+  return QStringLiteral( "checkgeometry" );
+}
+
+QString QgsGeometryCheckDegeneratePolygonAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm checks the degenerate polygons of a layer." );
```

Comment:
TBH, I have no idea what this means. I think we'll need to be a bit more verbose in these algs help text.